README.md

```diff
@@ -1,21 +1,121 @@
 # roave/you-are-using-it-wrong
 
-TBD
-
 [![Build Status](https://travis-ci.org/roave/you-are-using-it-wrong.svg?branch=master)](https://travis-ci.org/roave/you-are-using-it-wrong)
-[![Downloads](https://img.shields.io/packagist/dt/roave/you-are-using-it-wrong.svg)](https://packagist.org/packages/roave/you-are-using-it-wrong)
 [![Packagist](https://img.shields.io/packagist/v/roave/you-are-using-it-wrong.svg)](https://packagist.org/packages/roave/you-are-using-it-wrong)
-[![Dependencies](https://tidelift.com/badges/github/packagist/roave%2Fyou-are-using-it-wrong)](https://tidelift.com/subscription/pkg/packagist-roave%2Fyou-are-using-it-wrong?utm_source=packagist-ocramius%2Fpackage-versions&utm_medium=readme)
+
+This package enforces type checks during composer installation in downstream
+consumers of your package. This only applies to usages of classes, properties, methods and functions declared within packages that directly depend on *you-are-using-it-wrong*.  Issues that the static analyser finds that do not relate to these namespaces will not be reported.
+
+`roave/you-are-using-it-wrong` comes with a zero-configuration out-of-the-box
+setup.
+
+The usage of this plugin is highly endorsed for authors of new PHP libraries
+who appreciate the advantages of static types.
+
+This project is built with the hope that libraries with larger user-bases will
+raise awareness of type safety (or current lack thereof) in the PHP ecosystem.
+
+As annoying as it might sound, it is not uncommon for library maintainers to
+respond to support questions caused by lack of type checks in downstream
+projects. In addition to that, relying more on static types over runtime checks,
+it is possible to reduce code size and maintenance burden by strengthening the
+API boundaries of a library.
 
 ### Installation
 
-TBD
+This package is designed to be installed as a dependency of PHP **libraries**.
 
-### Use-cases
+In your library, add it to your
+`composer.json`:
 
-TBD
+```sh
+composer require roave/you-are-doing-it-wrong
+```
 
-## Professional Support
+No further changes are needed for this tool to start operating as per its
+design, if your declared types are already reflecting your library requirements.
+
+Please also note that this should **not** be used in `"require-dev"`, but
+specifically in `"require"` in order for the type checks to be applied to
+downstream consumers of your code.
+
+### Examples
+
+Assuming you have a library with following `composer.json`:
+
+```json
+{
+    "name": "my/awesome-library",
+    "type": "library",
+    "autoload": {
+        "psr-4": {
+            "My\\AwesomeLibrary\\": "src"
+        }
+    },
+    "require": {
+        "roave/you-are-doing-it-wrong": "@STABLE"
+    }
+}
+```
+
+Given following `src/MyHelloWorld.php`:
+
+```php
+<?php
```

Comment:
Declare strict?

Reply:
Wanted to keep the example minimal, but can do 👍

Reply:
that one's debatable. Good to encourage everyone to use declare strict, but it makes the motivation for this package a little less clear since strict would catch the type error shown. Could use a more complicated type like `string[]` in the example library and have the caller try to pass `int[]` then that may make the example over complicated.